src/Servers/Kestrel/Core/test/HttpRequestHeadersTests.cs

```diff
@@ -112,6 +113,23 @@ public void EntriesCanBeEnumeratedAfterResets()
         EnumerateEntries((IDictionary<string, StringValues>)headers);
     }
 
+    [Fact]
+    public void ClearPseudoRequestHeadersPlusResetClearsHeaderReferenceValue()
+    {
+        const BindingFlags privateFlags = BindingFlags.NonPublic | BindingFlags.Instance;
+
+        HttpRequestHeaders headers = new HttpRequestHeaders(reuseHeaderValues: false);
+        headers.HeaderMethod = "GET";
+        headers.ClearPseudoRequestHeaders();
+        headers.Reset();
+
+        // Hacky but required because header references is private.
+        var headerReferences = typeof(HttpRequestHeaders).GetField("_headers", privateFlags).GetValue(headers);
+        var methodValue = (StringValues)headerReferences.GetType().GetField("_Method").GetValue(headerReferences);
```

Comment:
@halter73 @Tratcher yolo?

The alternative is change the `HttpRequestHeaders.HeaderReferences` struct from private to internal and adding a method to get it on `HttpRequestHeaders`.

Reply:
*shrug* I don't care about reflection in a test.